src/components/CoursePage/LessonItem.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import LinkContainer from 'react-router-bootstrap/lib/LinkContainer';
+import withStyles from 'isomorphic-style-loader/lib/withStyles';
+import Glyphicon from 'react-bootstrap/lib/Glyphicon';
+import ListGroupItem from 'react-bootstrap/lib/ListGroupItem';
+import LevelIcon from '../LevelIcon';
+import PopoverComponent from '../PopoverComponent';
+import InstructionButton from '../InstructionButton';
+import Flag from '../Flag';
+import {createCheckboxesKey} from '../../util';
+import {getLessonFrontmatter} from '../../resources/lessonFrontmatter';
+import {getLevel} from '../../resources/lessons';
+import {getLessonIntro} from '../../resources/lessonContent';
+import {getTranslator} from '../../selectors/translate';
+import {onlyCheckedMainLanguage} from '../../selectors/filter';
+import {getNumberOfCheckedCheckboxes, getTotalNumberOfCheckboxes} from '../../selectors/checkboxes';
+import styles from './LessonItem.scss';
+
+
+const Progress = ({checkedCheckboxes, totalCheckboxes}) => {
+  return checkedCheckboxes > 0 ?
+    <div className={styles.progress}>
+      {`(${checkedCheckboxes}/${totalCheckboxes})`}
+    </div> :
+    null;
+};
+
+const Progressbar = ({checkedCheckboxes, totalCheckboxes, level}) => {
+  const progressPercent = totalCheckboxes > 0 ? 100 * checkedCheckboxes / totalCheckboxes : 0;
+  return level > 0 ?
+    <span className={styles['progressBarLevel' + level]} style={{width: progressPercent + '%'}}/> :
+    null;
+};
+
+const LessonItem = ({
+  course,
+  lesson,
+  language,
+  title,
+  level,
+  path,
+  external,
+  popoverContent,
+  isStudentMode,
+  onlyCheckedMainLanguage,
+  t,
+  checkedCheckboxes,
+  totalCheckboxes
+}) => {
+  const flag = !onlyCheckedMainLanguage ?
+    <div className={styles.flag}><Flag language={language}/></div> : null;
+
+  const instructionButton = isStudentMode ? null :
+    <InstructionButton {...{course, lesson, language, isReadme: true, onlyIcon: true, insideLink: true}} />;
+
+  const popoverButton = popoverContent ?
+    <PopoverComponent {...{popoverContent}}>
+      <Glyphicon className={styles.popoverGlyph} glyph='info-sign'/>
+    </PopoverComponent>
+    : null;
+
+  return (
+    <div>
+      {external ?
+        <ListGroupItem href={external} target="_blank" className={styles.row}>
+          {flag}
+          <LevelIcon level={level}/>
+          <div className={styles.title}>{title}</div>
+          <Glyphicon glyph="new-window"/>
+          <span className={styles.rightSide}>
+            {instructionButton}
```

Comment:
Vil man noen gang ha instructionButton for external?

Reply:
mulig... jeg lar den stå